ratatui-core/src/backend/test.rs

```diff
@@ -67,6 +68,225 @@ fn buffer_view(buffer: &Buffer) -> String {
+    }
+    if style.add_modifier.contains(crate::style::Modifier::DIM) {
+        tags.push("dim".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::CROSSED_OUT)
+    {
+        tags.push("crossed_out".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::SLOW_BLINK)
+    {
+        tags.push("slow_blink".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::RAPID_BLINK)
+    {
+        tags.push("rapid_blink".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::REVERSED)
+    {
+        tags.push("reversed".to_string());
+    }
+    if style.add_modifier.contains(crate::style::Modifier::HIDDEN) {
+        tags.push("hidden".to_string());
+    }
+
+    if tags.is_empty() {
+        String::new()
+    } else {
+        format!("{{{}}}", tags.join(","))
+    }
+}
+
+/// Format the end of a style tag based on the cell's style
+fn format_style_end(style: &crate::style::Style) -> String {
+    let mut tags = Vec::new();
+
+    // Add foreground color
+    if let Some(fg) = &style.fg {
+        if *fg != crate::style::Color::Reset {
+            tags.push(format!("/{}", format_color(*fg)));
+        }
+    }
+
+    // Add background color
+    if let Some(bg) = &style.bg {
+        if *bg != crate::style::Color::Reset {
+            tags.push(format!("/bg:{}", format_color(*bg)));
+        }
+    }
+
+    // Add modifiers
+    if style.add_modifier.contains(crate::style::Modifier::BOLD) {
+        tags.push("/bold".to_string());
+    }
+    if style.add_modifier.contains(crate::style::Modifier::ITALIC) {
+        tags.push("/italic".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::UNDERLINED)
+    {
+        tags.push("/underlined".to_string());
+    }
+    if style.add_modifier.contains(crate::style::Modifier::DIM) {
+        tags.push("/dim".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::CROSSED_OUT)
+    {
+        tags.push("/crossed_out".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::SLOW_BLINK)
+    {
+        tags.push("/slow_blink".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::RAPID_BLINK)
+    {
+        tags.push("/rapid_blink".to_string());
+    }
+    if style
+        .add_modifier
+        .contains(crate::style::Modifier::REVERSED)
+    {
+        tags.push("/reversed".to_string());
+    }
+    if style.add_modifier.contains(crate::style::Modifier::HIDDEN) {
+        tags.push("/hidden".to_string());
+    }
+
+    if tags.is_empty() {
+        String::new()
+    } else {
+        format!("{{{}}}", tags.join(","))
+    }
+}
+
+/// Format a color as a human-readable string
+fn format_color(color: crate::style::Color) -> String {
```

Comment:
Pretty sure we might have something like this already. Even though we don't, it should live in the color module.